WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/RefreshableScrollView.swift

```diff
@@ -0,0 +1,92 @@
+import SwiftUI
+
+/// A scroll view with pull-to-refresh support.
+struct RefreshableScrollView<Content: View>: View {
+    /// Callback that is triggered once refreshing completes.
+    typealias RefreshComplete = () -> Void
+
+    /// Refresh action that is called once refreshing starts.
+    /// `RefreshComplete` callback is triggered when refreshing completes.
+    typealias RefreshAction = (@escaping RefreshComplete) -> Void
+
+    /// Used to determine whether the spinner is shown or hidden.
+    @State private var isRefreshing: Bool = false
+
+    /// Keeps track of whether refresh action is enabled: it is enabled when the refresh action is not in progress.
+    /// Since `isRefreshing` boolean is set with animation when the spinner is shown/hidden, a separate boolean is needed.
+    @State private var canRefresh: Bool = true
+
+    private let refreshAction: RefreshAction
+    private let content: Content
+
+    // MARK: Constants
+
+    private let minOffsetToRefresh: CGFloat = 50.0
+
+    init(refreshAction: @escaping RefreshAction,
+         @ViewBuilder content: () -> Content) {
+        self.refreshAction = refreshAction
+        self.content = content()
+    }
+
+    var body: some View {
+        GeometryReader { geometry in
+            ScrollView {
+                VStack(spacing: 0) {
+                    if isRefreshing {
+                        ProgressView()
```

Comment:
nit: It would be great to maintain the same size of Orders and Products for the activity indicator:
```suggestion
                        ProgressView()
                        .scaleEffect(x: 1.5, y: 1.5, anchor: .center)
```

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/6272/commits/e428459c3abdd371bfe2f7b20f2e7d93ab6d7d23